index.bs

```diff
@@ -12595,7 +12610,70 @@ The \[[Prototype]] [=internal slot=] of an [=asynchronous iterator prototype obj
     1.  Return |object|'s [=default asynchronous iterator object/ongoing promise=].
 </div>
 
-Issue: <code>return</code>; <code>throw</code> methods?
+<div algorithm="to invoke the return property of asynchronous iterators">
+
+    If an [=asynchronous iterator return=] algorithm is defined for the [=inferface=], then the
+    [=asynchronous iterator prototype object=] must have a <code class="idl">return</code> data
+    property with attributes { \[[Writable]]: <emu-val>true</emu-val>,
+    \[[Enumerable]]: <emu-val>true</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
+    and whose value is a [=built-in function object=], taking one argument |value|, that behaves as
+    follows:
+
+    1.  Let |interface| be the [=interface=] for which the
+        [=asynchronous iterator prototype object=] exists.
+
+    1.  Let |returnPromiseCapability| be [=!=] [$NewPromiseCapability$]({{%Promise%}}).
+
+    1.  Let |object| be the result of calling [$ToObject$] on the
+        <emu-val>this</emu-val> value.
+
+    1.  [$IfAbruptRejectPromise$](|object|, |returnPromiseCapability|).
+
+    1.  If |object| [=is a platform object=], then [=perform a security check=], passing:
+        *   the platform object |object|,
+        *   the identifier "<code>return</code>", and
+        *   the type "<code>method</code>".
+
+        If this threw an exception |e|, then:
+        1.  Perform [=!=] [$Call$](|returnPromiseCapability|.\[[Reject]],
+            <emu-val>undefined</emu-val>, « |e| »).
+        1.  Return |returnPromiseCapability|.\[[Promise]].
+
+    1.  If |object| is not a [=default asynchronous iterator object=] for |interface|, then:
+        1.  Let |error| be a new {{ECMAScript/TypeError}}.
+        1.  Perform [=!=] [$Call$](|returnPromiseCapability|.\[[Reject]],
+            <emu-val>undefined</emu-val>, « |error| »).
+        1.  Return |returnPromiseCapability|.\[[Promise]].
+
+    1.  If |object|'s [=default asynchronous iterator object/ongoing promise=] is not undefined, then:
```

Comment:
Check if this undefined needs an emu-val? I don't remember which way this one goes.

Reply:
Mmm nice catch. Since we made it an IDL Promise, we should probably union it with null (another IDL value), not undefined (which is always `<emu-val>undefined</emu-val>`, a JS value). I switched all instances to do so.